odoo/http.py

```diff
@@ -1617,5 +1617,27 @@ def content_disposition(filename):
 
     return "attachment; filename*=UTF-8''%s" % escaped
 
+
+def set_header_field(headers, name, value):
+    """ Return new headers based on `headers` but with `value` set for the
+    header field `name`.
+
+    :param headers: the existing headers
+    :type headers: list of tuples (name, value)
+
+    :param name: the header field name
+    :type name: string
+
+    :param value: the value to set for the `name` header
+    :type value: string
+
+    :return: the updated headers
+    :rtype: list of tuples (name, value)
+    """
+    dictheaders = dict(headers)
+    dictheaders[name] = value
```

Comment:
I was reading https://github.com/odoo/odoo/pull/26635/#issuecomment-425043088 and I'm thinking this will be an issue for SVG specifically.

We probably need this function to do nothing if the given header is content-type and the given value is SVG?
Or is this function too low level, and there should be another layer to check the security when called from unsafe places?

@odony

Reply:
So creating a new function such as `set_safe_image_content_type` that sets it only if the target type is 'PNG', 'JPEG', 'GIF', 'ICO'?

Reply:
I have my suggested change ready if needed. Waiting on approval/pending reviews before force pushing.